titan-server/src/accounts/routes.rs

```diff
@@ -269,19 +269,26 @@ pub fn save_user_file_entry(
 /** **************************************************
  *  Excuses
  ** **************************************************/
+#[get("/excuses/unacknowledged")]
+pub fn list_unacknowledged_excuses(
+    titan_db: TitanPrimary,
+    wcf_db: UnksoMainForums,
+    app_config: State<config::AppConfig>,
+    auth_user: auth_guard::AuthenticatedUser
+) -> Json<Vec<models::UserEventExcuseWithAssoc>> {
+    Json(excuses::find_all_unacknowledged(
+        &*titan_db, &*wcf_db, &app_config).unwrap())
+}
 #[get("/<user_id>/excuses")]
 pub fn list_user_event_excuses(
     user_id: i32,
-    titan_primary: TitanPrimary,
+    titan_db: TitanPrimary,
+    wcf_db: UnksoMainForums,
+    app_config: State<config::AppConfig>,
     _auth_user: auth_guard::AuthenticatedUser
-) -> Result<Json<Vec<models::UserEventExcuseWithType>>, status::BadRequest<String>> {
-    let excuses_res = excuses::get_user_excuses(user_id, &*titan_primary);
-
-    if excuses_res.is_err() {
```

Comment:
We want to give informative errors to the client, but at the same time, not all internal errors need to be explained to the user (i.e. invalid values or unexpected state). If this query fails, a 500 response is sufficient for the client to infer the server was `unable to fetch excuses`, so I removed this message since it isn't useful.